File: core/network/src/test/kotlin/com/deepseek/harness/android/network/RpcEnvelopeTest.kt
package com.deepseek.harness.android.network

import kotlinx.serialization.encodeToString
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.buildJsonObject
import kotlinx.serialization.json.put
import org.junit.Assert.assertEquals
import org.junit.Test

class RpcEnvelopeTest {

    private val json = Json { ignoreUnknownKeys = true }

    @Test
    fun `client request keeps rpc identity`() {
        val request = ClientRequest(
            rpcId = "rpc-1",
            method = "session.prompt",
            payload = buildJsonObject { put("sessionId", "s1") },
        )

        val wire = json.encodeToString(request)
        val decoded = json.decodeFromString<ClientRequest>(wire)

        assertEquals(request.rpcId, decoded.rpcId)
        assertEquals(request.method, decoded.method)
    }
}

File: flutter/packages/network/test/rpc_envelope_test.dart
import 'dart:convert';

import 'package:test/test.dart';

import 'package:network/rpc_envelope.dart';

void main() {
  test('client request keeps rpc identity', () {
    final request = ClientRequest(
      rpcId: 'rpc-1',
      method: 'session.prompt',
      payload: <String, Object?>{'sessionId': 's1'},
    );

    final wire = jsonEncode(request.toJson());
    final decoded = ClientRequest.fromJson(jsonDecode(wire));

    expect(decoded.rpcId, request.rpcId);
    expect(decoded.method, request.method);
  });
}
